packages/forms-web-app/__tests__/unit/controllers/before-you-start/local-planning-department.test.js

```diff
@@ -0,0 +1,193 @@
+        selected: false,
+        text: undefined,
+        value: undefined,
+      },
+      {
+        selected: false,
+        text: 'lpa1',
+        value: 'lpa1',
+      },
+      {
+        selected: false,
+        text: 'lpa2',
+        value: 'lpa2',
+      },
+    ];
+
+    jest.resetAllMocks();
+  });
+
+  describe('Local Planning Department Controller Tests', () => {
+    it('Test the getPlanningDepartment method calls the correct template', async () => {
+      getRefreshedDepartmentData.mockResolvedValue(departmentsData);
+
+      appeal.lpaCode = '';
+      await localPlanningDepartmentController.getPlanningDepartment(req, res);
+
+      const { eligibleDepartments, ineligibleDepartments } = departmentsData;
+
+      expect(res.render).toBeCalledWith(VIEW.BEFORE_YOU_START.LOCAL_PLANNING_DEPARTMENT, {
+        appealLPD: '',
+        departments: departmentList,
+        eligibleDepartments,
+        ineligibleDepartments,
+      });
+    });
+
+    it('Test the getPlanningDepartment method calls the correct template', async () => {
+      getRefreshedDepartmentData.mockResolvedValue(departmentsData);
+      getDepartmentFromId.mockResolvedValue(undefined);
+
+      appeal.lpaCode = 'unknown';
+      await localPlanningDepartmentController.getPlanningDepartment(req, res);
+
+      const { eligibleDepartments, ineligibleDepartments } = departmentsData;
+
+      expect(res.render).toBeCalledWith(VIEW.BEFORE_YOU_START.LOCAL_PLANNING_DEPARTMENT, {
+        appealLPD: '',
+        departments: departmentList,
+        eligibleDepartments,
+        ineligibleDepartments,
+      });
+    });
+
+    it('Test the getPlanningDepartment method with existing LPD calls the correct template', async () => {
+      getRefreshedDepartmentData.mockResolvedValue(departmentsData);
+      getDepartmentFromId.mockResolvedValue({ name: 'lpdName' });
+
+      appeal.lpaCode = 'lpdCode';
+      await localPlanningDepartmentController.getPlanningDepartment(req, res);
+
+      const { eligibleDepartments, ineligibleDepartments } = departmentsData;
+
+      expect(res.render).toBeCalledWith(VIEW.BEFORE_YOU_START.LOCAL_PLANNING_DEPARTMENT, {
+        appealLPD: 'lpdName',
+        departments: departmentList,
+        eligibleDepartments,
+        ineligibleDepartments,
+      });
+    });
+
+    it('Test the postPlanningDepartment method call with handled department', async () => {
+      getRefreshedDepartmentData.mockResolvedValue(departmentsData);
+      getDepartmentFromName.mockResolvedValue({ id: 'lpaCode1', name: 'lpa1' });
+
+      const mockRequest = {
+        ...req,
+        body: { 'local-planning-department': 'lpa1' },
+      };
+
+      await localPlanningDepartmentController.postPlanningDepartment(mockRequest, res);
+
+      expect(createOrUpdateAppeal).toHaveBeenCalledWith({
+        ...appeal,
+        lpaCode: 'lpaCode1',
+      });
+
+      // expect(res.redirect).toBeCalledWith(`/${VIEW.}`); // Future Planning Application Decision Page
```

Comment:
Its better not to have commented out code. We can implement this when we play the future card

Reply:
Edited.

Reply:
Edited.

Reply:
Commented out code still exists